moler/device/textualdevice.py

```diff
@@ -130,6 +132,22 @@ def establish_connection(self):
         self._established = True
         self._log(level=logging.INFO, msg=msg)
 
+    def remove(self):
+        """
+        Closes device, if any command or event is attached to this device they will be finished.
+
+        :return: None
+        """
+        self._stop_prompts_observers()
+        if self.is_established():
+            self._established = False
+            self.io_connection.moler_connection.close()
+            self.io_connection.close()
+        super(TextualDevice, self).remove()
+        msg = "Device '{}' is closed.".format(self.name)
+        self._log(level=logging.INFO, msg=msg)
+        self._close_logger()
+
     def is_established(self):
```

Comment:
I would change naming. If I see `if device.is_established():` I don't know what it means. I need to jump into documentation then I see:
```
    def is_established(self):
        """
        Checks if connection is established to real device.
        :return: True if connection is established, False otherwise.
        """
```
Why not make it direct: `if device.has_established_connection():` or `if device.with_established_connection():`
Moreover, such naming build pair for `device.establish_connection()`

Reply:
Done